### PR TITLE
Dec 273 add 5 metadata fields

### PR DESCRIPTION
**Pull request by jonhartzler**
This is the first step in the metadata refactor.  I needed to update the meta data form to be more dynamic.  This change moves the form from rails to react.   The next step is to update the api to allow updating of items and for the react forms to show errors.  

I added a couple new generic components
1.  Form for a form
2.  FormRow
I also added StringField and TextField which are currently not in a generic space but attached to the MetaDataForm in the meta_data_form directory.  They may however be more generic then their current file placement suggests.  

Lastly and most important this pull request is going to go back into the phase2 branch not master.

### CODE REVIEW

**Comment by jaronkk** [comment]
I'm getting a lot of warnings and an error

1680x1050_-_cats_-_honeycomb

**Comment by jaronkk** [comment]
Making a change to the form and clicking the save button triggers a warning that I've made unsaved changes, and I have to click "Leave this page" to save the form.

File: app/assets/javascripts/components/meta_data_form/TextField.js.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  },
+
+  getDefaultProps: function() {
+    return {
+      value: ""
+    };
+  },
+
+  componentDidMount: function() {
+
```
Comment (rfox2) [inline app/assets/javascripts/components/meta_data_form/TextField.js.jsx]: Why the empty function?

File: app/assets/javascripts/components/meta_data_form/MetaDataForm.js.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    return (
+      <Panel PanelTitle="Meta Data">
+        <Form id="meta_data_form" url={this.props.url} authenticityToken={this.props.authenticityToken} method={this.props.method}>
+          <StringField name="item[name]" id="item_title" required="true" title="Title" value={this.state.formValues['item_title']} handleFieldChange={this.handleFieldChange} />
+          <TextField  name="item[description]" id="item_description" required="" title="Description" value={this.state.formValues['item_description']} handleFieldChange={this.handleFieldChange} />
+          <TextField  name="item[transcription]" id="item_transcription" required="" title="Transcription" value={this.state.formValues['item_transcription']} handleFieldChange={this.handleFieldChange} />
+          <StringField  name="item[manuscript_url]" id="item_manuscript_url" required="" title="Digitized Manuscript URL" value={this.state.formValues['item_manuscript_url']} handleFieldChange={this.handleFieldChange} placeholder="http://" help="Link to externally hosted manuscript viewer." />
+
+
+
```
Comment (rfox2) [inline app/assets/javascripts/components/meta_data_form/MetaDataForm.js.jsx]: Lots 'o' white space here. Necessary?

File: app/assets/javascripts/components/FormRow.js.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+//app/assets/javascripts/components/modal/Modal.jsx
+
+var FormRow = React.createClass({
+  displayName: 'Form',
+
+  propTypes: {
+    id: React.PropTypes.string.isRequired,
+    children: React.PropTypes.oneOfType([
```
Comment (rfox2) [inline app/assets/javascripts/components/FormRow.js.jsx]: I've not seen this syntax before. What does it mean?

Comment (jonhartzler) [inline app/assets/javascripts/components/FormRow.js.jsx]: Children are all the objects that are nested inside of another object

<FormRow>
  <p>Hello</p>
</FormRow>

<p>Hello</p> would be the children.

File: app/decorators/item_decorator.rb [hunk 4]
```diff
@@ -29,6 +29,20 @@ def show_image_box
     h.react_component "ItemShowImageBox", image: image_json, itemID: object.id.to_s
   end
 
+  def edit_form
```
Comment (jonhartzler) [inline app/decorators/item_decorator.rb]: This is not true.  Ruby cop is responding to the size of the method call inside this method.  There are no branching conditions.

File: app/decorators/item_decorator.rb [hunk 5]
```diff
@@ -29,12 +29,22 @@ def show_image_box
     h.react_component "ItemShowImageBox", image: image_json, itemID: object.id.to_s
   end
 
-  def edit_path
-    h.edit_item_path(object.id)
+  def edit_form
```
Comment (jonhartzler) [inline app/decorators/item_decorator.rb]: I think this is not true.  
Assignment Branch Condition matches branches it seems by measuring indents.  There are no branches just a complicated call to react component.

File: app/assets/javascripts/modified-check.js.coffee [hunk 6]
```diff
@@ -22,6 +22,7 @@ jQuery ->
 
 
   isDirty = () ->
+    return false
```
Comment (danthewolfe) [inline app/assets/javascripts/modified-check.js.coffee]: Add a comment here as to why it always returns false.

**Comment by danthewolfe** [comment]
:+1:

**Comment by jgondron** [comment]
:+1:

**Pull request merged**